src/shared/http/CurlHttpClient.php

```diff
@@ -121,8 +121,8 @@ private function setupCurlInstance(): void {
             $this->curl->setResolve($hostname . ':443:' . $this->config->getResolvedIp($hostname));
         }
 
-        if ($this->config->hasAuthenticationToken($hostname)) {
-            $headers[] = $this->config->getAuthenticationToken($hostname)->asString();
+        if ($this->config->hasAuthentication($hostname)) {
+            $headers[] = $this->config->getAuthentication($hostname)->asString();
```

Comment:
The only small thing I can think of here is that we might want to rename `asString` to something that makes it clear you'll get it back as a Http header line. So maybe `asHttpHeaderString()` (so we don't have to create an HttpHeader value object ;-) )